tools/istio-iptables/pkg/cmd/root.go

```diff
@@ -83,7 +84,7 @@ var rootCmd = &cobra.Command{
 			}
 		}
 		if cfg.RunValidation {
-			hostIP, _, err := getLocalIP()
+			hostIP, _, err := getLocalIP(DualStack || cfg.DualStack)
```

Comment:
Hi @adams-shaun, one question for it: shall we uniform both the environment variable `ISTIO_DUAL_STACK` and cfg configuration together? which means that it may be easier for us to use only one assertion condition. And thanks for this PR!

Reply:
I agree, this part of the code was smelly.  I was originally trying to use root_test to test both for enabling the ENABLE_INBOUND_IPV6 config option.  

1. `ISTIO_DUAL_STACK` already present for execution of istio-iptables in the init container
2. For istio-cni, this needed to be piped in via the Redirect (followed other examples.)

The confusing bit was the `TestGetLocalIP` was calling `constructConfig` directly.  This makes it difficult to isolate the inputs.   I think I've cleaned it up a fair bit and kept the coverage.  Let me know what you think.

Reply:
Hi @adams-shaun, there is a test flag `isDS` in `TestGetLocalIP` and I also set `DualStack = tt.isDS` at L#94, so is it possible for you to change code to fix the issue that `TestGetLocalIP` directly calls `constructConfig`? And thanks for your clarification.